smart-contracts/contracts/PublicLock.sol

```diff
@@ -221,20 +221,42 @@ contract PublicLock is ILockCore, ERC165, IERC721, IERC721Receiver, Ownable {
   }
 
   /**
-   * @dev Called by owner to wiwthdraw all funds from the lock.
+   * @dev Called by owner to withdraw all funds from the lock.
    * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
-   * TODO: consider partial withdraws?
    * TODO: check for re-entrency?
    */
-  function withdraw(
-  )
+  function withdraw()
     external
     onlyOwner
   {
     uint256 balance = address(this).balance;
     require(balance > 0, "Not enough funds");
+    _withdraw(balance);
+  }
+
+  /**
+   * @dev Called by owner to partially withdraw funds from the lock.
+   * TODO: consider allowing anybody to trigger this as long as it goes to owner anyway?
+   * TODO: check for re-entrency?
+   */
+  function partialWithdraw(uint _amount)
+    external
+    onlyOwner
+  {
+    uint256 balance = address(this).balance;
+    require(balance > 0 && balance > _amount, "Not enough funds");
+    _withdraw(_amount);
+  }
+
+  /**
+   * @dev private version of the withdraw function which handles all withdrawals from the lock.
+   * TODO: check for re-entrency?
+   */
+  function _withdraw(uint _amount)
+    private
+  {
     address owner = Ownable.owner();
-    owner.transfer(balance);
+    owner.transfer(_amount);
```

Comment:
Should we include an event here?

Reply:
Yes! Actually, events, in general, is on my list of things to look at. I think this should have one, and there may be other areas where we want to add events as well. For now I'll focus on this one

Reply:
Julien had mentioned possible opening this function up to be called be addresses other than the owner in the future, and we know the transfer can only go to the owner so I'm thinking of this:
```
event Withdrawal(
    address _sender,
    uint _amount
    );
```

Reply:
with `address _sender` possibly indexed?

Reply:
Having other people trigger a withdraw seems a bit strange to me.  And if we allow cancels (https://github.com/unlock-protocol/unlock/issues/637) others could troll the system by calling withdraw, leaving not enough money in the contract to allow refunds.  If we include this feature, we may want to consider permissions for it.

But yea, if we include that feature, the event looks good.  Yes include `indexed` here.

Reply:
Yes, you echoed my concerns around this exactly. I thought it would require a registry of whitelisted addresses or some other such permissions to prevent that exact attack.

Reply:
done.

Reply:
I think you both are right. I just wanted a way for "us" to trigger withdraws in order to make sure we keep the locks "safe" (funds are safer on a user's address than inside a smart contract IMO)... but maybe this is not the right approach. Maybe insisting on them withdrawing is the way forward. Let's push that to the backlog and see how 'disciplined' people are.

Reply:
@julien51 Would you like me to remove `address _sender` from the event for now, and only add it in if we end up implementing that feature?

Reply:
That is probably a good idea! Thanks!